libs/language-server/src/lib/ast/model-util.ts

```diff
@@ -209,15 +216,68 @@ export function inferTypesFromValue(
   if (isValuetypeAssignmentLiteral(value)) {
     return [PropertyValuetype.VALUETYPE_ASSIGNMENT];
   }
-  if (isCollectionLiteral(value)) {
-    return [PropertyValuetype.COLLECTION];
-  }
   if (isConstraintReferenceLiteral(value)) {
     return [PropertyValuetype.CONSTRAINT];
   }
+
   assertUnreachable(value);
 }
 
+export function evaluateExpression(expression: BooleanExpression): boolean {
+  if (isBooleanLiteral(expression)) {
+    return expression.value;
+  }
+  if (isUnaryExpression(expression)) {
+    const unaryOperator = expression.operator;
+    switch (unaryOperator) {
+      case 'not': {
+        return !evaluateExpression(expression.expression);
+      }
+      default:
+        assertUnreachable(unaryOperator);
+    }
+  }
+  if (isBinaryExpression(expression)) {
+    const binaryOperator = expression.operator;
+    switch (binaryOperator) {
+      case '==': {
```

Comment:
Imho these should be constants, not strings.

Reply:
Note that they are not hardcoded. Instead, they are compliant with the literal type of the generated AST.

Excerpt from the generated `ast.ts`:
``` typescript
export interface BinaryExpression extends AstNode {
    readonly $container: BinaryExpression | CollectionLiteral | PropertyAssignment | UnaryExpression | ValuetypeDefinition;
    readonly $type: 'BinaryExpression';
    left: BooleanExpression
    operator: '!=' | '==' | 'and' | 'or' | 'xor'
    right: BooleanExpression
}
```

Reply:
Not really the only thing enums would be good for but in that case I don't feel bad about doing it like this, makes sense 👍